### PR TITLE
Moved dirs for expected CODON hierarchy - Centralised shared config.

### PR DESCRIPTION
**Pull request by marcoooo**

### CODE REVIEW

File: modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    return {
+        %{$self->SUPER::default_options},
+        'base_dir'    => $self->o('ENV', 'BASE_DIR'),
+        'env_user'    => $self->o('ENV', 'USER'),
+        'species'     => [],
+        'antispecies' => [],
+        'division'    => [],
+        'mart_dir'    => getcwd,
+        'scratch_dir' => catdir('/hps', 'scratch', $self->o('env_user'), $self->o('pipeline_name')),
+        'test_dir'    => catdir($self->o('scratch_dir'), 'mart_test'),
```
[the remark this answers is not in the document]
Comment (marcoooo) [reply modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm:63]: I agree for the `one week` and the potential need to review it afterwards, but having it in a far more "permanent" dir will surely end up with very old directories staying there for ever.

File: modules/Bio/EnsEMBL/PipeConfig/BuildGeneMartMTMPTables_conf.pm [hunk 2]
```diff
@@ -23,130 +22,113 @@ package Bio::EnsEMBL::PipeConfig::BuildGeneMartMTMPTables_conf;
...
-    scripts_dir              => $self->o('base_dir').'/ensembl-variation/scripts/',
-    variation_import_lib     => $self->o('scripts_dir').'import',
-    variation_feature_script => $self->o('scripts_dir').'misc/mart_variation_effect.pl',
-    variation_mtmp_script    => $self->o('scripts_dir').'misc/create_MTMP_tables.pl',
-    mart_phenotypes_script   => $self->o('scripts_dir').'misc/mart_phenotypes.pl',
-  };
+    my ($self) = @_;
+    return {
+        %{$self->SUPER::default_options},
+        run_all                  => 0,
```
[the remark this answers is not in the document]
Comment (marcoooo) [reply modules/Bio/EnsEMBL/PipeConfig/BuildGeneMartMTMPTables_conf.pm]: https://github.com/Ensembl/ensembl-biomart/pull/145/commits/a615f5d04cc2c732510f75292f0db4b0abc33d2d

File: modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    return {
+        'default' => { 'LSF' => '-q production' },
+        'low'     => { 'LSF' => '-q production -M 2048 -R "rusage[mem=2048]"' },
+        'mem'     => { 'LSF' => '-q production -M 8192 -R "rusage[mem=8192]"' },
+        'himem'   => { 'LSF' => '-q production -M 16384 -R "rusage[mem=16384]"' }
+    };
+}
+
+
+sub pipeline_wide_parameters {
```
[the remark this answers is not in the document]
Comment (marcoooo) [reply modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm]: https://github.com/Ensembl/ensembl-biomart/pull/145/commits/1ad32858192b346cb1518154fc10abdf5a631e8e

File: modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcoooo) [reply modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm]: https://github.com/Ensembl/ensembl-biomart/pull/145/commits/24f70ae429eb1228a2526357854ff9d1da0499f7

File: modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    my ($self) = @_;
+    return {
+        %{$self->SUPER::default_options},
+        'base_dir'    => $self->o('ENV', 'BASE_DIR'),
+        'env_user'    => $self->o('ENV', 'USER'),
+        'species'     => [],
+        'antispecies' => [],
+        'division'    => [],
+        'mart_dir'    => getcwd,
+        'scratch_dir' => catdir('/hps', 'scratch', $self->o('env_user'), $self->o('pipeline_name')),
```
[the remark this answers is not in the document]
Comment (marcoooo) [reply modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm]: https://github.com/Ensembl/ensembl-biomart/pull/145/commits/a6c41c747e5b47976cd792589118544ed697907a

File: modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (james-monkeyshines) [reply modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm]: Yes - my thoughts on that were that we could have a post-release cleanup task. Also, when updating the SOPs for Codon execution, I've been adding instructions to tidy up, e.g. https://www.ebi.ac.uk/seqdb/confluence/display/GTI/Xref+pipeline#Xrefpipeline-Tidyingup.

[the remark this answers is not in the document]
Comment (marcoooo) [reply modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm]: Cheers. We are on the same pace, was thinking about updating SOP when starting release. such as we clean up FTP dirs as well

**Review by james-monkeyshines — APPROVED** [review]

**Pull request merged**